### PR TITLE
Set up devbox for a consistent dev/CI environment

### PR DESCRIPTION
**Pull request by K-Phoen**
To ensure a consistent experience when working on cog, this PR introduces devbox as a way of providing the tools required by cog.

### CODE REVIEW

**Review by spinillos — APPROVED** [review]

File: devbox.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      "echo 'Installing dependencies...' && make deps"
+    ],
+    "scripts": {
+      "go-example":  "make run-go-example",
+      "java-example": "make run-java-example",
+      "php-example": "make run-php-example",
+      "ts-example":  "make run-ts-example",
+      "lint":        "make lint",
+      "tests":       "make tests",
+      "gen-sdk-dev": "make gen-sdk-dev"
```
Comment (spinillos) [inline devbox.json:27]: Python is missing on purpose?

Comment (K-Phoen) [reply devbox.json:27]: For some reason we didn't run the python examples in the CI before. I didn't fix that in this PR, I'll do it in another one if we decide to merge this :)

Comment (spinillos) [reply devbox.json:27]: I tested the rest of them and they are working :)

File: README.md [hunk 2]
```diff
@@ -36,7 +36,36 @@ For the [`kind-registry`](https://github.com/grafana/kind-registry):
...
+```console
+$ devbox run
+```
+
+Packages can be installed using:
+
+```console
+devbox add go@1.23
+```
+
```
Comment (spinillos) [inline README.md:70]: Maybe add `exit` to exit from the shell.

Comment (K-Phoen) [reply README.md:72]: Done :)

**Pull request merged**